src/domain/interfaces/prices-api.interface.ts

```diff
@@ -8,23 +8,11 @@ export interface IPricesApi {
     fiatCode: string;
   }): Promise<AssetPrice>;
 
-  getTokenPrice(args: {
+  getTokenPrices(args: {
     chainName: string;
-    tokenAddress: string;
+    tokenAddresses: string[];
     fiatCode: string;
-  }): Promise<AssetPrice>;
+  }): Promise<AssetPrice[]>;
 
   getFiatCodes(): Promise<string[]>;
-
-  /**
-   * Registers a price reference for token address as not-found.
-   * This function allows the clients of this interface to signal that a price couldn't
-   * be parsed from its {@link AssetPrice}. This allows the underlying implementation
-   * to delay subsequent requests targeting the same token address.
-   */
-  registerNotFoundTokenPrice(args: {
```

Comment:
This function is not used anymore since the clients of this interface don't need to signal the absence of a price anymore. (This is not encapsulated in the CoingeckoAPI implementation itself).

If we agree on the current state of this PR, `ICacheService.expire` won't be needed anymore as well, but I'm leaving that change to be done in a separated PR based on that outcome.